### PR TITLE
Add errorCount to file/list response

### PR DESCRIPTION
**Pull request by shashigharti**
Read errorCount and add it to file/list response

- fixes #[135](https://github.com/frictionlessdata/application/issues/135)

---

Please make sure that all the checks pass. Please add here any additional information regarding this pull request. It's highly recommended that you link this PR to an issue (please create one if it doesn't exist for this PR)

### CODE REVIEW

File: frictionless/project/database.py [hunk 1]
```diff
@@ -201,15 +201,21 @@ def list_records(self) -> List[IRecordItem]:
                     self.records.c.type,
                     self.records.c.updated,
                     self.records.c.tableName,
+                    self.records.c.report,
                 )
             )
             items: List[IRecordItem] = []
             for row in result:
+                errorCount = 0
+                if row.report:
```
Comment (roll) [inline frictionless/project/database.py]: Can you please add a TODO that it's very ineffective (parsing report every time) and we need to add `errorCount` as a database column in the future?

File: frictionless/server/file/list.py [hunk 2]
```diff
@@ -16,5 +16,14 @@ class Result(BaseModel):
 @router.post("/file/list")
 def server_file_list(request: Request, props: Props) -> Result:
     project: Project = request.app.get_project(props.session)
+    records = project.database.list_records()
     items = project.list_files()
+    for index, item in enumerate(items):
```
Comment (roll) [inline frictionless/server/file/list.py]: This logic needs to go to `project.list_files` as we don't use endpoints for any actual code ;)

**Comment by roll** [comment]
@shashigharti 
I added a few comments but it's good in-general

**Comment by shashigharti** [comment]
@roll Thank you! I will make the required changes.

File: frictionless/project/database.py (hunk 1 above)
Comment (shashigharti) [reply frictionless/project/database.py]: Added here https://github.com/frictionlessdata/application/issues/146

**Comment by shashigharti** [comment]
@roll Thank you! and it is ready for the review!

File: frictionless/project/interfaces.py [hunk 3]
```diff
@@ -11,6 +11,7 @@
 class IFileItem(TypedDict, total=False):
     path: Required[str]
     type: str
+    errorCount: int
```
Comment (roll) [inline frictionless/project/interfaces.py:14]: @shashigharti 
I removed here `Optional` as `total=False` in `TypedDict` means everything is optional unless it's marked as `Required` - https://peps.python.org/pep-0655/

Comment (roll) [reply frictionless/project/interfaces.py:14]: @shashigharti 
And yea actually, it led to updating the actual logic as for `file.errorCount` we should have 3 states:
- undefined
- 0 (valid)
- X (invalid)

and we can't have `None` here

**Review by roll — APPROVED** [review]
Thanks!

File: frictionless/project/interfaces.py (hunk 3 above)
Comment (shashigharti) [reply frictionless/project/interfaces.py:14]: Thank you!. I was making it complex :(

Comment (roll) [reply frictionless/project/interfaces.py:14]: Not at all it's just very confusing topic (missing values) especially when we need to pair Python and JavaScript

**Pull request merged**